contracts/src/CrossChainApplications/examples/NativeTokenBridge/tests/NativeTokenDestinationTests.t.sol

```diff
@@ -87,7 +92,72 @@ contract NativeTokenDestinationTest is NativeTokenBridgeTest {
             allowedRelayerAddresses: new address[](0),
             message: abi.encode(
                 ITokenSource.SourceAction.Unlock,
-                abi.encode(_DEFAULT_RECIPIENT, _DEFAULT_TRANSFER_AMOUNT)
+                abi.encode(_DEFAULT_RECIPIENT, _DEFAULT_TRANSFER_AMOUNT * _DEFAULT_TOKEN_MULTIPLIER)
+                )
+        });
+
+        vm.expectCall(
+            MOCK_TELEPORTER_MESSENGER_ADDRESS,
+            abi.encodeCall(ITeleporterMessenger.sendCrossChainMessage, (expectedMessageInput))
+        );
+
+        nativeTokenDestination.transferToSource{value: _DEFAULT_TRANSFER_AMOUNT}(
+            _DEFAULT_RECIPIENT,
+            TeleporterFeeInfo({feeTokenAddress: address(mockERC20), amount: _DEFAULT_FEE_AMOUNT}),
+            new address[](0)
+        );
+    }
+
+    function testTransferToSourceDivideOnSend() public {
+        nativeTokenDestination = new NativeTokenDestination({
+            teleporterRegistryAddress: MOCK_TELEPORTER_REGISTRY_ADDRESS,
+            teleporterManager: _DEFAULT_OWNER_ADDRESS,
+            sourceBlockchainID_: _DEFAULT_OTHER_CHAIN_ID,
+            nativeTokenSourceAddress_: _DEFAULT_OTHER_BRIDGE_ADDRESS,
+            initialReserveImbalance_: _DEFAULT_INITIAL_RESERVE_IMBALANCE,
+            tokenMultiplier_: _DEFAULT_TOKEN_MULTIPLIER,
+            multiplyOnSend_: false
+        });
+
+        vm.expectEmit(true, true, true, true, address(nativeTokenDestination));
+        emit CollateralAdded({amount: _DEFAULT_INITIAL_RESERVE_IMBALANCE, remaining: 0});
+
+        // We shouldn't mint anything here.
+        vm.expectCall(
+            NATIVE_MINTER_PRECOMPILE_ADDRESS,
+            abi.encodeWithSelector(INativeMinter.mintNativeCoin.selector),
+            0
+        );
+
+        vm.prank(MOCK_TELEPORTER_MESSENGER_ADDRESS);
+        nativeTokenDestination.receiveTeleporterMessage(
```

Comment:
can you add a unit test that just checks the emit `NativeTokensMinted` for a `receiveTeleoprterMessage` where some amount is used as collateral, and the rest is minted.